subprojects/logging/src/main/java/org/gradle/internal/logging/console/ProgressOperation.java

```diff
@@ -65,7 +65,21 @@ public ProgressOperation getParent() {
         return parent;
     }
 
-    public Set<ProgressOperation> getChildren() {
-        return children;
+    public boolean addChild(ProgressOperation operation) {
+        if (children == null) {
+            children = new HashSet<ProgressOperation>();
+        }
+        return children.add(operation);
+    }
+
+    public boolean removeChild(ProgressOperation operation) {
+        if (children == null) {
```

Comment:
I don't think so. We are not enforcing that `operation` exist in children, but we could by saying `children == null || !children.contains(operation)` without a high cost